core/ecschema-metadata/src/Metadata/Class.ts

```diff
@@ -146,18 +146,27 @@ export abstract class ECClass extends SchemaItem implements CustomAttributeConta
    * @param excludeInherited If true, excludes inherited properties from the results. Defaults to false.
    */
   public async getProperty(name: string, excludeInherited: boolean = false): Promise<Property | undefined> {
+    const upperKey = name.toUpperCase();
+
+    if(this._mergedPropertyCache?.has(upperKey)) {
```

Comment:
This is probably a bug if excludeInherited is set to true, because the cache will also return inherited properties